lib/input_sanitizer/errors.rb

```diff
@@ -30,6 +30,16 @@ def initialize
     end
   end
 
+  class RegexpMismatchError < ValidationError
+    def code
+      :regexp
+    end
+
+    def initialize(regexp)
+      super("does not match the regular expression: #{regexp}")
```

Comment:
I am not 100% sold on the fact of exposing our validation regexp to the client. IMO this is too much of an implementation detail that we use a regexp. I think this is case by case, e.g. we could use regexp for validating UUID format and then the error message sent to the user should be "not a valid UUID".

What is your regexp use case? I'm wondering wheter you have something that exposing the regexp to the user of th API would make sens.

Reply:
I am not sure who do you consider a _client_ in this sense. This is a gem so I assume that the client is the application using it. These errors should not be exposed further.

In my use case, I validate the input in a controller and if it's invalid return HTTP/422 without any details. We don't want to give a hint of what is wrong in this case. Even if we wanted, still I think that we should return something that makes sense in the context of the service not simply return validation errors to the client.

What do you think?

Reply:
Perhaps I'm mistaken but I think the errors are often forewarded in responses as-is. As we agreed offline a good compromise would be not exposing the regexp cause in practice the use of the gem should tweak the message to suite the business case (e.g. "uuid invalid" in the mentioned example)

Reply:
Sure, let's not expose the regexp.